embassy-net/src/lib.rs

```diff
@@ -363,6 +366,55 @@ impl<D: Driver + 'static> Stack<D> {
         v4_up || v6_up
     }
 
+    /// Wait for the network stack to obtain a valid IP configuration.
+    ///
+    /// ## Notes:
+    /// - Ensure [`Stack::run`] has been called before using this function.
+    ///
+    /// - This function may never yield (e.g. if no configuration is obtained through DHCP).
```

Comment:
```suggestion
    /// - This function may never return (e.g. if no configuration is obtained through DHCP).
```

"yield" in async means the future suspending on an .await, not returning.

Reply:
good catch, thanks